naga/src/back/msl/mod.rs

```diff
@@ -530,6 +530,11 @@ impl Options {
                     crate::BuiltIn::PrimitiveIndex if self.lang_version < (2, 2) => {
                         return Err(Error::UnsupportedAttribute("primitive_id".to_string()));
                     }
+                    // macOS: Since Metal 2.2
+                    // iOS: Since Metal 2.3 (check depends on https://github.com/gfx-rs/naga/issues/2164)
+                    crate::BuiltIn::Barycentric if self.lang_version < (2, 2) => {
```

Comment:
The check should be conservative if it doesn't have the information it needs. If you could also change the test for primitive id, that would be good.

Reply:
this is the exact same comment as above and the exact same check as above, it is also non-conservative and outdated issue link